### PR TITLE
[US3.3.3] Register Device + [3.3.4] Unlink Device + includes main class implementation

### PR DESCRIPTION
**Pull request by nugyn**
This pull request includes, the register device, unlink device, and a main class as entry point into the arduino funnel into pi to aws iot

### CODE REVIEW

**Comment by nugyn** [comment]
Make sure we Squash these commits, to make development look a little cleaner

File: functionality/pub.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+- subscribe()
+- publish ()
+'''
+
+class Publisher:
+    pub = ""
+    broker_address = ""
+    port = ""
+    def __init__(self):
+        self.broker_address = "110.174.81.168"
```
Comment (auspham) [inline functionality/pub.py]: store IP in configuration/ constant file?

File: functionality/register.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+    '''
+    main methods
+    - connect()
+    - disconnect()
+    - subscribe()
+    - publish ()
+    '''
+
+    broker_address = "110.174.81.168"
```
Comment (auspham) [inline functionality/register.py]: store IP in configuration/ constant file?

Comment (auspham) [reply functionality/register.py]: and port

File: functionality/sub.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
+import paho.mqtt.client as mqtt
+import time as time
+import json
+
+user = "user"
+password = "password"
+topic_1 = "awsiot_to_localgateway"
+topic_2 = "both_directions"
+broker_address="110.174.81.168"
```
Comment (auspham) [inline functionality/sub.py]: store IP & port in configuration/ constant file?

File: functionality/unlinkdevice.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+import register
+import json
+import iddevice
+
+'''
+Unlink Device, by publishing ID to Cloud, and it will find in the dynamoDB database and update by removing it
+'''
+
+broker_address = "110.174.81.168"
+port = 1883
```
Comment (auspham) [inline functionality/unlinkdevice.py]: store IP & port in configuration/ constant file?

**Review by auspham — CHANGES_REQUESTED** [review]

File: functionality/unlinkdevice.py (hunk 4 above)
Comment (ijkeleher) [reply functionality/unlinkdevice.py]: Modified to load from .env and added bash script to generate .env

**Comment by ijkeleher** [comment]
@rockmanvnx6 

python files now use dotenv to pull values from config file .env

script to be run at setup:

```
#!/bin/bash

IP="$(curl icanhazip.com)"

echo "BROKER_IP=$IP
BROKER_PORT=1883" > .env```

**Review by ijkeleher — APPROVED** [review]

File: functionality/pub.py (hunk 1 above)
Comment (nugyn) [reply functionality/pub.py]: Inci did the pleasure of giving me the implementation of a .env file @inci90  thanks!

File: functionality/sub.py (hunk 3 above)
Comment (nugyn) [reply functionality/sub.py]: I will create a json file with all topics

**Pull request merged**